### PR TITLE
Sync: Update full sync to expand theme data before sending it on full…

### PR DESCRIPTION
**Pull request by enejb**
Expand the theme data at the before sending it on full sync. This ensure that the full sync only send the most up to date data.

### CODE REVIEW

**Comment by gravityrail** [comment]
Refactored a bit to address feedback

File: tests/php/sync/test_class.jetpack-sync-full.php [hunk 1]
```diff
@@ -686,7 +686,7 @@ function test_full_sync_doesnt_send_deleted_users() {
 
 	function test_full_sync_status_with_a_small_queue() {
 
-		$this->sender->set_dequeue_max_bytes( 1500 ); // process 0.0015MB of items at a time\
+		$this->sender->set_dequeue_max_bytes( 700 ); // process 0.0007MB of items at a time
```
Comment (gravityrail) [inline tests/php/sync/test_class.jetpack-sync-full.php]: What's the reason for this change?

Comment (enejb) [inline tests/php/sync/test_class.jetpack-sync-full.php]: The reason for that change is that the test was failing. Since now the number of bytes is smaller pre expand.

**Comment by gravityrail** [comment]
Merged to 4.2
